src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -48,10 +48,10 @@ public TypeFactory(OutputLibrary library, Type unknownType)
             InputListType { IsEmbeddingsVector: true } listType => new CSharpType(typeof(ReadOnlyMemory<>), listType.IsNullable, CreateType(listType.ElementType)),
             InputListType listType => new CSharpType(typeof(IList<>), listType.IsNullable, CreateType(listType.ElementType)),
             InputDictionaryType dictionaryType => new CSharpType(typeof(IDictionary<,>), inputType.IsNullable, typeof(string), CreateType(dictionaryType.ValueType)),
-            InputEnumType enumType => _library.ResolveEnum(enumType).WithNullable(inputType.IsNullable),
+            InputEnumType enumType => _library.ResolveEnum(enumType).IsNullable == inputType.IsNullable ? _library.ResolveEnum(enumType) : _library.ResolveEnum(enumType).WithNullable(inputType.IsNullable),
```

Comment:
we could just simplify this as 
```suggestion
            InputEnumType enumType => _library.ResolveEnum(enumType).WithNullable(inputType.IsNullable),
```
because `WithNullable` has the check if `isNullable` is the same as the nullable value we have in this type, return `this`